src/components/radio-group/radio-group.tsx

```diff
@@ -75,67 +82,85 @@ export class RadioGroup extends React.Component<Partial<RadioGroupProps>, {}> {
             this.checkedArray.forEach(data => {
                 data.checked = false;
             });
             this.checkedArray[index].checked = true;
             if (this.props.onChange) {
-                this.props.onChange(value);
+                this.props.onChange(e);
             }
         };
     }
 
-    private createChildrenFromDataSource(): React.ReactNode[] {
-        const childArray: React.ReactNode[] = [];
-        if (Array.isArray(this.props.dataSource)) {
-            this.props.dataSource!.forEach((data, index) => {
-                const props: RadioButtonProps = {
-                    key: index,
-                    value: data.value,
-                    automationId: 'RADIO_BUTTON_' + index,
-                    checked: this.checkedArray[index].checked,
-                    onClick: this.childrenOnClick(index),
-                    disabled: this.props.disabled || data.disabled,
-                    location: this.props.location,
-                    name: this.name
-                };
-                childArray.push(React.createElement(RadioButton, props));
-            });
-        }
-        return childArray;
+    private createChildrenFromDataSource(): Array<React.ReactElement<RadioButton>> {
+        const isGroupChecked: boolean = this.isGroupChecked();
+
+        return this.props.dataSource!.map((props, index) => (
+            <RadioButton
+                key={index}
+                value={props.value}
+                data-automation-id={'RADIO_BUTTON_' + index}
+                checked={this.checkedArray[index].checked}
+                onChange={this.childrenOnClick(index)}
+                disabled={this.props.disabled || props.disabled}
+                location={this.props.location}
+                name={this.name}
+                className="radioGroupChild"
+                tabIndex={this.getChildTabIndex(index, isGroupChecked)}
+            />
+        ));
     }
 
-    private createChildren(dataArray: any): React.ReactNode[] {
-        const childArray: React.ReactNode[] = [];
-        for (let index = 0; index < dataArray.length; index++) {
-            const data = dataArray[index];
-
-            if (data.type === RadioButton) {
-                const props: RadioButtonProps = {
-                    key: index,
-                    value: data.props.value,
-                    automationId: 'RADIO_BUTTON_' + index,
-                    checked: this.checkedArray[index].checked,
-                    onClick: this.childrenOnClick(index),
-                    disabled: this.props.disabled || data.props.disabled,
-                    location: this.props.location,
-                    name: this.name};
-
-                childArray.push(React.cloneElement(data as React.ReactElement<any>, props));
+    private createChildren(dataArray: React.ReactNode[]): React.ReactNode[] {
+        const isGroupChecked: boolean = this.isGroupChecked();
+
+        return dataArray.map((data, index) => {
```

Comment:
dataArray is this.props.children? Then shouldn't its type be React.ReactNode?
If dataArray is ReactNode maybe use React.Children.map?
Then you don't need to use "element = data as ..."

Reply:
Agree regarding using `React.Children.map`. Regarding the element casting is needed since the interface is: `type ReactNode = ReactChild | ReactFragment | boolean | null | undefined;` that means that `element.type` might not exist on `boolean | null | undefined` for instance